torch/onnx/symbolic_helper.py

```diff
@@ -1114,27 +1114,43 @@ def _avgpool_helper(tuple_fn, padding, kernel_size, stride, divisor_override, na
     return padding
 
 
-def check_training_mode(op_train_mode, op_name):
-    op_train_mode = True if op_train_mode == 1 else False
-    if GLOBALS.training_mode is not None and op_train_mode != GLOBALS.training_mode:
-        op_mode = "training " if op_train_mode else "inference"
-        training_mode = "training " if GLOBALS.training_mode else "inference"
-        # setting the model mode could result in op_mode != _flags.training_mode
-        # if the model is a FuncModule. In this case we warn the user of
-        # the state and export depending on op_mode
-        # This is to support use-cases of fixing certain layer weights
-        # in training.
-        warnings.warn(
-            "ONNX export mode is set to "
-            + training_mode
-            + " mode, but operator "
-            + op_name
-            + " is set to "
-            + op_mode
-            + " mode. The operators will be exported in "
-            + op_mode
-            + ", as specified by the functional operator."
-        )
+def check_training_mode(op_train_mode: int, op_name: str) -> None:
+    """Warns the user if the model's training mode and the export mode do not agree."""
+    if GLOBALS.training_mode == _C_onnx.TrainingMode.PRESERVE:
+        # Do not modify the training mode
+        return
+
+    if op_train_mode:
+        op_mode_enum = _C_onnx.TrainingMode.TRAINING
+    else:
+        op_mode_enum = _C_onnx.TrainingMode.EVAL
+    if op_mode_enum == GLOBALS.training_mode:
+        # The modes agree. Do nothing
+        return
+
+    assert GLOBALS.training_mode in {
+        _C_onnx.TrainingMode.TRAINING,
+        _C_onnx.TrainingMode.EVAL,
+    }, "Bug: training_mode should only be 'TRAINING' or 'EVAL' at this point"
+
+    export_mode_text = (
+        "training"
+        if GLOBALS.training_mode == _C_onnx.TrainingMode.TRAINING
+        else "inference"
+    )
+    op_mode_text = (
```

Comment:
Let's use PyTorch terminology for this to make the relationship with the user's code clearer. In PyTorch, users call `module.train(True)` or `module.train(False)`.

```suggestion
    op_mode_text = f"train={bool(op_train_mode)}"
```

Reply:
Done